lib/safer_rails_console/patches/sandbox/auto_rollback.rb

```diff
@@ -4,9 +4,20 @@ module Sandbox
       module AutoRollback
         module ActiveRecord
           module ConnectionAdapters
-            module AbstractAdapter
+            module AbstractAdapter4
               def log(sql, name = 'SQL', binds = [], statement_name = nil)
-                super(sql, name, binds, statement_name) { yield }
+                super
+              rescue => e
+                connection = ::ActiveRecord::Base.connection
+                connection.rollback_db_transaction
+                connection.begin_db_transaction
+                raise e
```

Comment:
Consider extracting these 4 lines into a method like `rollback_and_begin_new_transaction!` that you can use in both `AbstractAdapter` versions.